### PR TITLE
Add BaseCPS

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Adds the BaseCPS dataset.

### CODE REVIEW

**Comment by nikhilwoodruff** [comment]
@MaxGhenis @nmrodelo - I've added an example for both the survey variable file and the input variable file (interest from the CPS INT_VAL field, into interest_income in openfisca-us). The openfisca-us variable doesn't actually exist in openfisca-us yet so would need to be added (without the formula).

**Comment by nikhilwoodruff** [comment]
Actually, I'm just going to merge this to get the openfisca-us PR tests running.

File: openfisca_data/datasets/us/cps/base_cps/dataset.py [hunk 1]
```diff
@@ -1,10 +1,92 @@
...
+
+        person["person_weight"] = person.P_A_FNLWGT
+        tax_unit = pd.DataFrame(index=person.person_tax_unit_id.unique())
+        spm_unit = pd.DataFrame(index=person.person_spm_unit_id.unique())
+        tax_unit["tax_unit_id"] = tax_unit.index
+        spm_unit["spm_unit_id"] = spm_unit.index
+        tax_unit[
+            "tax_unit_weight"
+        ] = 1e5  # not accurate, just a placeholder until development
+        spm_unit["spm_unit_weight"] = 1e5
```
Comment (MaxGhenis) [inline openfisca_data/datasets/us/cps/base_cps/dataset.py:78]: SPM weight is in the file

File: openfisca_data/datasets/us/cps/base_cps/dataset.py [hunk 2]
```diff
@@ -1,10 +1,92 @@
...
+        household["household_id"] = household.H_H_SEQ
+
+        person["person_weight"] = person.P_A_FNLWGT
+        tax_unit = pd.DataFrame(index=person.person_tax_unit_id.unique())
+        spm_unit = pd.DataFrame(index=person.person_spm_unit_id.unique())
+        tax_unit["tax_unit_id"] = tax_unit.index
+        spm_unit["spm_unit_id"] = spm_unit.index
+        tax_unit[
+            "tax_unit_weight"
+        ] = 1e5  # not accurate, just a placeholder until development
```
Comment (MaxGhenis) [inline openfisca_data/datasets/us/cps/base_cps/dataset.py:77]: Does every level need a weight?

File: openfisca_data/datasets/us/cps/base_cps/dataset.py [hunk 3]
```diff
@@ -1,10 +1,92 @@
...
+        person["person_household_id"] = person.P_PH_SEQ
+        person["person_family_id"] = person.P_PH_SEQ * 10 + person.P_PF_SEQ
+        person["person_id"] = person.P_PH_SEQ * 100 + person.P_P_SEQ
+        person["person_tax_unit_id"] = person.P_TAX_ID
+        person["person_spm_unit_id"] = person.P_SPM_ID
+
+        family["family_id"] = family.F_FH_SEQ * 10 + family.F_FFPOS
+        household["household_id"] = household.H_H_SEQ
+
+        person["person_weight"] = person.P_A_FNLWGT
```
Comment (MaxGhenis) [inline openfisca_data/datasets/us/cps/base_cps/dataset.py:70]: Needs to be `MARSUPWT`

File: openfisca_data/datasets/us/cps/base_cps/dataset.py (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_data/datasets/us/cps/base_cps/dataset.py:77]: No, we can just return Microseries with weights set to 1.

File: openfisca_data/datasets/us/cps/base_cps/dataset.py (hunk 1 above)
Comment (nikhilwoodruff) [reply openfisca_data/datasets/us/cps/base_cps/dataset.py:78]: Added - SPM_Weight

File: openfisca_data/datasets/us/cps/base_cps/dataset.py (hunk 3 above)
Comment (nikhilwoodruff) [reply openfisca_data/datasets/us/cps/base_cps/dataset.py:70]: Changed

**Pull request merged**